cupy/_environment.py

```diff
@@ -10,6 +10,8 @@
 import sys
 import warnings
 
+import pkg_resources
```

Comment:
Do you know whether the import is fast enough in the Python versions CuPy supports?

https://github.com/cupy/cupy/blob/9704573e333760ec80cb7353eef66ed94bf4afab/cupy/testing/_helper.py#L1252-L1254

Reply:
Good point! This was added from the user on Windows. https://github.com/cupy/cupy/pull/2759

`pkg_resources` is part of `setuptools`, so it can be depending on the `setuptools` version too.
I'm not 100% sure about this import is performant in all environment, so maybe it's better to avoid using `pkg_resources`. I'm thinking of `importlib.metadata` instead. Unfortunately, it is only supported in Python 3.8+.

Reply:
```python
installed = {d.metadata['Name'] for d in importlib.metadata.distributions()}
```
or
```python
cupy_installed = [name for name in known if list(importlib.metadata.distributions(name=name))]
```
if `importlib.metadata` is used.

Reply:
I'm thinking of using `importlib.metadata` for Python 3.8+ and do nothing for 3.7 or earlier.
[`importlib-metadata`](https://pypi.org/project/importlib-metadata/) package is available for backport for 3.7 but we don't want to introduce additonal dependency.